pkg/injector/test/test_helpers.go

```diff
@@ -0,0 +1,102 @@
+const directoryForYAMLFiles = "test_fixtures"
+
+var log = logger.New("sidecar-injector")
+
+// LoadExpectedEnvoyYAML loads the expectation for a given test from the file system. This must run within ginkgo.It()
+func LoadExpectedEnvoyYAML(filename string) string {
+	filePath := path.Join(directoryForYAMLFiles, filename)
+
+	// The filename will contain the name of the function by convention
+	log.Info().Msgf("Loading expected test output from file %s", filename)
+
+	if _, err := os.Stat(filepath.Clean(filePath)); os.IsNotExist(err) {
+		log.Warn().Msg("File at path %s did not exist; Going to create it")
+		if createdFile, err := os.Create(filepath.Clean(filePath)); err != nil {
+			log.Err(err).Msgf("Error creating file for expected output of test: %s", filePath)
+		} else {
+			_ = createdFile.Close()
+		}
+	}
+	expectedEnvoyConfig, err := ioutil.ReadFile(filepath.Clean(filePath))
+	if err != nil {
+		log.Err(err).Msgf("Error reading expected Envoy bootstrap YAML from file %s", filename)
+	}
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	return string(expectedEnvoyConfig)
+}
+
+// SaveFile saves the given content into the file name (within a directory dedicated for these tests). This must run within ginkgo.It()
+func SaveFile(fileName string, fileContent []byte) {
+	filePath := path.Join(directoryForYAMLFiles, fileName)
+	err := ioutil.WriteFile(filepath.Clean(filePath), fileContent, 0600)
+	if err != nil {
+		log.Err(err).Msgf("Error writing actual Envoy Cluster XDS YAML to file %s", fileName)
+	}
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+}
+
+// MarshalAndSaveToFile converts a generic Go struct into YAML and saves it to a file. This must run within ginkgo.It()
+func MarshalAndSaveToFile(someStruct interface{}, filename string) string {
+	yamlBytes, err := yaml.Marshal(someStruct)
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	SaveFile(filename, yamlBytes)
+	return string(yamlBytes)
+}
+
+// ThisFunction runs the given function in a ginkgo.Context(), marshals the output and compares to an expectation loaded from file.
+func ThisFunction(functionName string, fn func() interface{}) {
+	ginkgo.Context(fmt.Sprintf("ThisFunction %s", functionName), func() {
+		ginkgo.It("creates Envoy config", func() {
+			expectedFilename := fmt.Sprintf("expected_output_%s.yaml", functionName)
+			actualFilename := fmt.Sprintf("actual_output_%s.yaml", functionName)
+
+			actual := fn()
+
+			expectedYAML := LoadExpectedEnvoyYAML(expectedFilename)
+			actualYAML := MarshalAndSaveToFile(actual, actualFilename)
+
+			Compare(functionName, actualFilename, expectedFilename, actualYAML, expectedYAML)
+		})
+	})
+}
+
+// Compare is a wrapper around gomega.Expect().To(Equal()) and compares actualYAML and expectedYAML; It also provides a verbose message when things don't match with a tip on how to fix things.
+func Compare(functionName, actualFilename, expectedFilename, actualYAML, expectedYAML string) {
```

Comment:
a diff based compare would produce more readable output. Something like https://github.com/andreyvit/diff or whatever you are comfortable to use (as long as the same is used across all the code)

Reply:
@khenidak I agree this returns too much in console, given that we also save the actual and the expected to disk.

I removed showing actual & expected.  Will only show delta.

Speaking of delta - we could but we don't need to use @andreyvit's 'diff' (which I love by the way - looked at it at some point) - the Gomega framework gives us the delta.  Take a look at the screenshot below, where I swapped a 0 w/ 1:

![image](https://user-images.githubusercontent.com/49918230/108284471-23f17280-713a-11eb-9009-7e4086e248ff.png)